typescript/api/test/unit/services/FigshareService.test.ts

```diff
@@ -0,0 +1,135 @@
+const { describe, it, beforeEach } = require('mocha');
+const { expect } = require('chai');
+
+// Minimal sails stub required by FigshareService
+(global as any).sails = {
+  config: {
+    figshareAPIEnv: {
+      overrideArtifacts: {
+        mapping: {
+          artifacts: {}
+        }
+      }
+    },
+    figshareAPI: {
+      APIToken: 'test-token',
+      baseURL: 'https://api.figshare.test',
+      frontEndURL: 'https://figshare.example',
+      mapping: {
+        upload: {
+          fileListPageSize: 2,
+          override: {}
+        },
+        figshareOnlyPublishSelectedAttachmentFiles: false,
+        recordAllFilesUploaded: false,
+        targetState: {},
+        response: {
+          article: {}
+        }
+      },
+      diskSpaceThreshold: 0
+    },
+    figshareReDBoxFORMapping: {
+      FORMapping: []
+    },
+    record: {
+      createUpdateFigshareArticleLogLevel: 'verbose'
+    },
+    queue: {
+      serviceName: ''
+    }
+  },
+  log: {
+    verbose: () => {},
+    info: () => {},
+    error: () => {},
+    warn: () => {},
+    debug: () => {}
+  },
+  services: {},
+  on: () => {}
+};
+
+// axios is required at module load in FigshareService, so we must set the stub before importing it
+const axiosCalls: any[] = [];
+let axiosResponses: any[] = [];
+const axiosStub = (config) => {
+  axiosCalls.push(config);
+  if (!axiosResponses.length) {
+    return Promise.reject(new Error('No axios mock response available'));
+  }
+  return Promise.resolve(axiosResponses.shift());
+};
+(require as any).cache[require.resolve('axios')] = { exports: axiosStub as any };
+
+const { Services } = require('../../../services/FigshareService');
+const FigshareService = Services.FigshareService;
+
+describe('FigshareService - getArticleFileList pagination', () => {
```

Comment:
I'd like to see a test that returns no files, to make sure the paging works in that situation.